rapid7_vulndb/komand_rapid7_vulndb/actions/search_db/action.py

```diff
@@ -1,37 +1,31 @@
 import komand
 from .schema import SearchDbInput, SearchDbOutput, Input, Output
 # Custom imports below
-from komand_rapid7_vulndb.util import path_helper, browser
+from typing import Dict, List
+import requests
+from komand_rapid7_vulndb.util import utils
 
 
 class SearchDb(komand.Action):
 
     def __init__(self):
         super(self.__class__, self).__init__(
-                name='search_db',
-                description='Search the database to find vulnerabilities and exploits',
-                input=SearchDbInput(),
-                output=SearchDbOutput())
+            name='search_db',
+            description='Search the database to find vulnerabilities and exploits',
+            input=SearchDbInput(),
+            output=SearchDbOutput())
 
     def run(self, params={}):
         # Get params
         search = params.get(Input.SEARCH)
         data_base = params.get(Input.DATABASE)
 
-        search = search.replace(" ", "+")
-        self.logger.info(f'Searching for {search}')
-
-        # Set database xpath
-        temp = path_helper.set_xpath(data_base)
-        xpath = temp['xpath']
-        db = temp['db']
-        self.logger.info('Database set')
-
-        # Create web browser
-        base_url = f'https://www.rapid7.com/db/search?utf8=%E2%9C%93&q={search}&t={db}'
-        vuldb_browser = browser.VulnDBBrowser(data_base)
-        results = vuldb_browser.scrape_vuldb(base_url, xpath)
-        if results['results']:
-            return {Output.SEARCH_RESULTS: results['results'], Output.RESULTS_FOUND: results['found']}
-        else:
-            return {Output.RESULTS_FOUND: results['found']}
+        q = {"query": search}
+        q = utils.R7VDB.set_query_db_type(q, data_base)
+        data = utils.R7VDB.get_query(q)
+        num_of_pages = data['metadata']['total_pages']
+        results = utils.R7VDB.paginate_search(q, num_of_pages)
```

Comment:
Could we simply call `search` or `paginate_search` without need of `get_query` to get number of total pages?

Reply:
sure, I've moved the pagination part to utils